### PR TITLE
Split pipelines into various flavours.

### PR DESCRIPTION
**Pull request by tmcgilchrist**
Split deployer into the following pipelines:
* Tarides - existing Tarides/OCamlLabs pipelines on deploy.ci3.ocamlabs.io
* OCaml - new pipelines for deploy.ci.ocaml.org
* Toxis - existing Mirage piplines on deploy.ocamllabs.io

f37c1b75d820a84238e470f9aa19f14d75ab321f adds v2.ocaml.org deployer
67ffbeef09ba4dffb408cebcec9fce02066ab8d5 adds new opam-3.ocaml.org based off a new docker setup  @ https://github.com/tmcgilchrist/opam2web/tree/updates

TODO
- [x] Install https://github.com/apps/deploy-ci-ocaml-org into [ocaml/v2.ocaml.org](https://github.com/ocaml/v2.ocaml.org),  [ocaml/ocaml.org](https://github.com/ocaml/ocaml.org) and [ocaml-opam/opam2web](https://github.com/ocaml-opam/opam2web)

### CODE REVIEW

File: src/pipeline_ocaml_org.ml [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+  | `Linux_x86_64
+]
+
+let pool_id : arch -> string = function
+  | `Linux_arm64 -> "linux-arm64"
+  | `Linux_x86_64 -> "linux-x86_64"
+
+module Cluster = struct
+  (* Strings here represent the docker context to use. *)
+  module Ocamlorg_docker = Current_docker.Make(struct let docker_context = Some "TODO" end)
```
Comment (talex5) [inline src/pipeline_ocaml_org.ml:38]: `TODO`?

**Review by talex5 — COMMENTED** [review]
What's the goal here? This seems to be duplicating a lot of code for no obvious reason. I would have thought that only the `v` function would need to be different for each deployer.

`local` already lets you build only a single target, and flavours seem like just a slight extension of that.

**Comment by tmcgilchrist** [comment]
The goal is to have 2 main versions of deployer; one for deploy.ci3.ocamllabs.io and one for deploy.ci.ocaml.org.
Everything for ocaml.org goes onto the deployer under that domain, and everything else stays on ci3.
When the solver service is done, we can migrate the deployer off toxis and retire that GH App
I wanted to avoid having different pipelines on the live branches vs what is present on master.

This means there are now 3 GH Apps:
 * deploy.ci.ocaml.org @ https://github.com/apps/deploy-ci-ocaml-org
 * deploy.ci3.ocamllabs.io (confusingly called ocaml.org deployer) @ https://github.com/apps/ocaml-org-deployer
 * toxis deployer (OCurrent deployer) @ https://github.com/apps/ocurrent-deployer

File: src/pipeline_ocaml_org.ml (hunk 1 above)
Comment (tmcgilchrist) [reply src/pipeline_ocaml_org.ml]: Added as a new docker context.

**Pull request merged**